src/app/not-found.tsx

```diff
@@ -13,7 +13,8 @@ export default function Page() {
     name: "graphql.github.io",
   }
 
-  const title = `Found broken \`${mounted ? pathname?.replace(/\/$/, "") : ""}\` link. Please fix!`
+  const referrer = document?.referrer
+  const title = `Found broken \`${mounted ? pathname?.replace(/\/$/, "") : ""}\` link${referrer ? ` from \`${referrer}\`` : ""}. Please fix!`
```

Comment:
```suggestion
  const title = `Found broken \`${mounted ? pathname?.replace(/\/$/, "") : ""}\` link${referrer ? ` from \`${document.referrer}\`` : ""}. Please fix!`
```

to avoid optional chaining